webpack/assets/javascripts/react_app/components/Editor/EditorHelpers.js

```diff
@@ -0,0 +1,157 @@
+  const returns = `Returns:\n\t${methodReturns(method)}`;
+  if (isEmpty(method.examples))
+    return [method.short_description, usage, returns].join('\n\n');
+
+  let examples = method.examples
+    .map(e => {
+      const desc = e.desc ? `${e.desc}\n\n` : '';
+      return `${desc}${e.example}`;
+    })
+    .join('\n\n');
+  examples = `Examples:\n\n${examples}`;
+  return [method.short_description, usage, returns, examples].join('\n\n');
+};
+
+const methodSignature = (method, isSnippet) => {
+  if (isEmpty(method.params)) return method.name;
+
+  let currParam = 1;
+  const params = method.params.map(p => {
+    switch (p.type) {
+      case 'required':
+        return isSnippet ? `\${${currParam++}:${p.name}}` : p.name;
+      case 'optional': {
+        if (p.expected_type === 'list') {
+          return isSnippet
+            ? `\${${currParam++}:first},\${${currParam++}:second}`
+            : `*${p.name}`;
+        }
+        const defaultVal = resolveDefault(p);
+        return isSnippet
+          ? `\${${currParam++}:${defaultVal}}`
+          : `${p.name} = ${defaultVal}`;
+      }
+      case 'keyword':
+        return isSnippet
+          ? `${p.name}: \${${currParam++}:${resolveDefault(p)}}`
+          : `${p.name}: ${resolveDefault(p)}`;
+      default:
+        return null;
+    }
+  });
+  const filtered = filter(params, p => p !== null).join(', ');
+  const blockParam = find(method.params, p => p.type === 'block');
+  const parts = [method.name];
+  if (!isEmpty(method.params)) parts.push(`(${filtered})`);
+  if (blockParam) parts.push(` ${blockParam.schema}`);
+  return parts.join('');
+};
+
+export const parseDocs = cache => {
+  // Do nothing if the editor is not being used for templates
+  if (!cache) return;
+  // Custom basic snippets
+  /* eslint-disable */
+  let completions = [
+    {
+      caption: '<%',
+      snippet: '<% ${1:code} %>',
+      tooltip: 'Executes code, but does not insert a value\n\n<% code %>',
+      meta: 'erb',
+    },
+    {
+      caption: '<%=',
+      snippet: '<%= ${1:expression} %>',
+      tooltip: 'Inserts the value of an expression\n\n<%= expression %>',
+      meta: 'erb',
+    },
+    {
+      caption: '<%-',
+      snippet: '<% ${1:code} -%>',
+      tooltip:
+        'Executes code, but does not insert a value, trims the following line break\n\n<% code -%>',
+      meta: 'erb',
+    },
+    {
+      caption: '<%=-',
+      snippet: '<%= ${1:expression} -%>',
+      tooltip:
+        'Inserts the value of an expression, trims the following line break\n\n<%= expression -%>',
+      meta: 'erb',
+    },
+    {
+      caption: '<%#',
+      snippet: '<%# ${1:comment} -%>',
+      tooltip:
+        'Comment, removed from the final output, trims the following line break\n\n<%# comment -%>',
+      meta: 'erb',
+    },
+  ];
+  /* eslint-enable */
+  // Parse JSON with DSL documentation
+  forEach(JSON.parse(cache).docs.classes, cls => {
```

Comment:
If you try to create a new template with empty content, you'll get Rails validation error but the Editor will disappear with following error in console:
```
Uncaught TypeError: Cannot read properties of null (reading 'docs')
    at parseDocs (EditorHelpers.js:135:11)
    at EditorActions.js:73:3
    at Object.dispatch (index.js:8:1)
    at dispatch (<anonymous>:3665:80)
    at redux.js:477:1
    at Editor.componentDidMount (Editor.js:44:5)
    at commitLifeCycles (react-dom.development.js:19814:1)
    at commitLayoutEffects (react-dom.development.js:22803:1)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188:1)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237:1)
```

Reply:
That's actually a great find! This also was reproducible via edition.